### PR TITLE
fuzz: Port correctness/cse fuzzer over to libfuzzer

### PR DESCRIPTION
**Pull request by silvergasp**

### CODE REVIEW

**Review by steven-johnson — APPROVED** [review]

File: test/fuzz/cse.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#include <fuzzer/FuzzedDataProvider.h>
+#include <time.h>
+
+using namespace Halide;
+using namespace Halide::Internal;
+using std::vector;
+
+Expr random_expr(FuzzedDataProvider &fdp, int depth, vector<Expr> &exprs) {
+    if (depth <= 0) {
+        return fdp.ConsumeIntegralInRange<int>(-5, 4);
```
Comment (silvergasp) [inline test/fuzz/cse.cpp:12]: Looks fine #7546 origional `rng() % 10 - 5`

File: test/fuzz/cse.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    if (!exprs.empty() && fdp.ConsumeBool()) {
+        // Reuse an existing expression
+        return exprs[fdp.ConsumeIntegralInRange<size_t>(0, exprs.size() - 1)];
```
Comment (silvergasp) [inline test/fuzz/cse.cpp:17]: Looks fine, explicitly references size()-1

File: test/fuzz/cse.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    Expr next;
+    switch (fdp.ConsumeIntegralInRange<int>(0, 8)) {
```
Comment (silvergasp) [inline test/fuzz/cse.cpp:21]: This looks ok.

**Review by silvergasp — COMMENTED** [review]
#7546 These all look fine

File: test/fuzz/cse.cpp (hunk 3 above)
Comment (steven-johnson) [reply test/fuzz/cse.cpp:21]: If the idea is that a value of 8 is intended to hit the default case, we should add a comment to that effect

File: test/fuzz/cse.cpp (hunk 2 above)
Comment (steven-johnson) [reply test/fuzz/cse.cpp:17]: Or better yet, use PickValueInArray()?

Comment (silvergasp) [reply test/fuzz/cse.cpp:17]: PickValueInArray doesn't work on vectors... see [implementation](https://github.com/llvm/llvm-project/blob/main/compiler-rt/include/fuzzer/FuzzedDataProvider.h#L75-L78) although this is probably annoying enough that I might just create a PR with the LLVM project.

File: test/fuzz/cse.cpp (hunk 3 above)
Comment (silvergasp) [reply test/fuzz/cse.cpp:21]: yeah that's what it looks like, I was mostly just attempting to replicate the original logic but while I'm at it I can probably make this clearer too.

File: test/fuzz/cse.cpp (hunk 2 above)
Comment (steven-johnson) [reply test/fuzz/cse.cpp:17]: (But could always just make a local `PickValueInVector()` wrapper to use in these tests, nice for clarity)

**Pull request merged**